### PR TITLE
Fix to handle large png frames

### LINKED ISSUE #13: Bug handling large PNG files containing multiple data chunks in one frame
There is a rather serious issue when handling large PNG files at the moment. The following code:

```python
from apng import APNG

im = APNG.open("test_large.apng")
im.frames[0][0].save("frame0.png")
```
generates broken png files for individual frames, if a frame contain more that one data chunk.
cf. #12 for a fix.

### PR DESCRIPTION
**Pull request by bpotard**
Fixes #13.

This PR fixes a bug where individual PNG frames were not valid when saved as individual PNG files. This was traced to a bug in the APNG load code, where the "head_chunks" were added before _each_ data chunk, instead of just once per frame. This only affected "large" PNG files, i.e. frames that didn't fit in a single data chunk.

### CODE REVIEW

**Comment by eight04** [comment]
We need a test case.

**Comment by bpotard** [comment]
Sure!
```shell
wget http://littlesvr.ca/apng/images/Contact.png
```

Then in python:
```python
from apng import APNG
from wand.image import Image
im = APNG.open("Contact.png")
for i, (png, control) in enumerate(im.frames):
  png.save("{i}.png".format(i=i))

Image(filename="0.png")
```
You get:
```
wand.exceptions.CoderError: Not enough image data `0.png' @ error/png.c/MagickPNGErrorHandler/1643
```
The file is 65411 bytes.
The file cannot be displayed by firefox and probably some other browsers, chrome displays half of the image; pillow appears fine with it?

With my fix:
0.png is 64618 bytes.
It can be opened by wand, firefox, etc.

Hope that helps!

File: apng/__init__.py [hunk 1]
```diff
@@ -443,6 +443,7 @@ def from_bytes(cls, b):
 		frame_chunks = []
 		frames = []
 		num_plays = 0
+		new_frame = True
```
Comment (eight04) [inline apng/__init__.py]: We should rename this flag to `frame_has_head_chunks = False`.

File: apng/__init__.py [hunk 2]
```diff
@@ -458,7 +459,7 @@ def from_bytes(cls, b):
 					# IDAT inside chunk, go to next frame
 					frame_chunks.append(end)
 					frames.append((PNG.from_chunks(frame_chunks), control))
-					
+					new_frame = True
```
Comment (eight04) [inline apng/__init__.py]: This line should be inserted after L467 and changed to `frame_has_head_chunks = False`.

File: apng/__init__.py [hunk 3]
```diff
@@ -467,11 +468,15 @@ def from_bytes(cls, b):
 				else:
 					control = FrameControl.from_bytes(data[12:-4])
 			elif type_ == "IDAT":
-				frame_chunks.extend(head_chunks)
+				if new_frame:
```
Comment (eight04) [inline apng/__init__.py]: `if not frame_has_head_chunks:`, etc.

**Review by eight04 — CHANGES_REQUESTED** [review]
I have added the disassemble test to the master branch and confirmed it works. Would you like to add the assemble test? You need to extract each frame to the fixture folder:
https://github.com/eight04/pyAPNG/tree/master/test/fixtures/multi_fdat

Here is an example:
https://github.com/eight04/pyAPNG/tree/master/test/fixtures/ball

**Comment by bpotard** [comment]
All the requested changes have now been done, and it appears the tests are passing fine.

**Review by eight04 — APPROVED** [review]

**Comment by eight04** [comment]
:+1: Any idea why we have such messy diffs on Github? It looks fine in the terminal with `git diff master` command.

**Pull request merged**